consensus/phases/data.go

```diff
@@ -219,3 +254,49 @@ type DeviantBitSet struct {
 	HighBitLength uint8
 	Payload       []byte
 }
+
+type Phase2Packet struct {
+	// -------------------- Header
+	packetHeader PacketHeader
+
+	// -------------------- Section 1
+	globuleHashSignature    [64]byte
+	deviantBitSet           DeviantBitSet
+	signatureHeaderSection1 [64]byte
+
+	// -------------------- Section 2 (optional)
+	votesAndAnswers         []ReferendumVote
+	signatureHeaderSection2 [64]byte
+}
+
+func (p2p *Phase2Packet) isPhase3Needed() bool {
+	return p2p.packetHeader.f00
+}
+
+func (p2p *Phase2Packet) hasSection2() bool {
+	return p2p.packetHeader.f01
+}
+
+func (p2p *Phase2Packet) SetPacketHeader(header *RoutingHeader) error {
+	if header.PacketType != types.Phase2 {
+		return errors.New("Phase2Packet.SetPacketHeader: wrong packet type")
+	}
+
+	p2p.packetHeader.setRoutingFields(header, NetworkConsistency, 2)
+
+	return nil
+}
+
+func (p2p *Phase2Packet) GetPacketHeader() (*RoutingHeader, error) {
+	header := &RoutingHeader{}
+
+	if p2p.packetHeader.PacketT != NetworkConsistency {
+		return nil, errors.New("Phase2Packet.GetPacketHeader: wrong packet type")
+	}
+
+	header.PacketType = types.Phase2
+	header.OriginID = p2p.packetHeader.OriginNodeID
```

Comment:
p2p name can confuse. Maybe better FirstPhasePacket, SecondPhasePacket and fpp spp for recievers